### ISSUE TITLE
NVENC codec now fails to record or stream.

### ISSUE THREAD

**Issue by OoLunar** [obsproject/obs-studio#6138]
### Operating System Info

Other

### Other OS

Void Linux x64

### OBS Studio Version

Git

### OBS Studio Version (Other)

OBS Studio - 27.2.0-101-g358ef281c-modified (linux)

### OBS Studio Log URL

https://obsproject.com/logs/V50esYgas7cGcRHg

### OBS Studio Crash Log URL

N/A

### Expected Behavior

To stream/record as expected.

### Current Behavior

![image](https://user-images.githubusercontent.com/46751150/158048140-10551ca6-2cf5-46de-a9bf-7c8058ab9e72.png)
![image](https://user-images.githubusercontent.com/46751150/158048156-87099c73-5446-42db-8b8f-dc0e157d1f3f.png)

The dropdown above shows that NVENC is indeed a recognized and available encoder.

### Steps to Reproduce

Compiled on latest commit. This bug does not occur 30 commits ago (random number chosen)

### Anything else we should know?

Compiled with the following:
```
cmake .. -DUNIX_STRUCTURE=0 -DENABLE_PIPEWIRE=OFF -DCEF_ROOT_DIR=/home/lunar/Code/Repos/obs/cef_binary_4638_linux64 -DCMAKE_INSTALL_PREFIX=/home/lunar/.local/share/obs-studio/ -DCMAKE_BUILD_TYPE=Release -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON -DCMAKE_CXX_FLAGS=-march=native

make -j$(nproc)
make install
```

If any other information is required, don't hesitate to ask.

**Comment by OoLunar** [comment]
Possibly linked to #6109

**Comment by username-murphy** [comment]
Hardware / Software issues
 require
Hardware / Software information
to generate
Hardware / Software solutions.

The correctness and quality of the Hardware / Software solutions generated is almost always directly indicative to the level of detail and accuracy provided about the Hardware / Software issues.

ie: sh#t_in_sh#t_out

* Issue RFI    : Operating System Info
Your answer: Other
Are we supposed to guess?

Just a suggestion how to answer this.

Type of Install: Linux System Wide, Dual Boot, VM
How is the install designed to work: GPU Passthrough? Prime Render Offload?
Platform:          Desktop or Laptop or Raspi
CPU? :             AMD, Intel ?
Mboard:           Make and model if its a desktop.?
Graphics card: nVidia, Intel AMD / Make and Model?
Graphics Driver: Whats the release? where from?

* Other OS
Your answer:   Void Linux x64
What? release? kernel? Where from?

Desktop: kDE, Plasma, XFCE, i3
Compositor: Kwin Sway. none

Any relevent system custom configs?
Have you proved the successfull optimal functioning of the System? Graphics Driver?

* OBS Studio Crash Log URL
Your answer:   N/A
This isnt optional. 
Your asking the OBS Dev and others for free community support but you could make the effort to attach the " OBS Studio Crash Log URL" .  An OBS Dev might find it slightly useful.

**Comment by OoLunar** [comment]
While I entirely understand that my bug "report" isn't entirely verbose, there's no reason to get so hostile about it. Simply asking kindly for more details is all that's needed. "Hey, we're going to need a bit more information on your hardware. Could you provide us with your CPU and GPU, what arch it is, etc" Additionally, "other" was a dropdown from their Bug Report layout, as shown below, and I tried to follow the format of it:
![image](https://user-images.githubusercontent.com/46751150/158064864-eb544446-458c-4d16-adf9-68c3995ce144.png)

> Your asking the OBS Dev and others for free community support but you could make the effort to attach the " OBS Studio Crash Log URL" . An OBS Dev might find it slightly useful

There wasn't a crash log because this wasn't a crash. Believe me, I'm not trying to make the devs break their backs trying to help me, I'm just not sure what other information they need.

> Did you look at the log you did attach?

Yes, I did. I believe the issue here was the invalid encoder ID which failed to render correctly in the terminal. As such, I don't think I could've fixed that, which was part of why I came here.

> Does AMD Ryzen 5 3550H with Radeon Vega Mobile Gfx run NVENC ?

No, my integrated GPU does not have NVENC support. However my GeForce GTX 1650 Mobile / Max-Q does, which has been used previously to record.

> My best guess is your system is far from being completely, absolutely and correctly configured into / onto and with your hardware platform.

I kindly ask for you to not look down on other people so quickly.

> Have you proved the successfull optimal functioning of the System?

> Compiled on latest commit. This bug does not occur 30 commits ago

I wouldn't have opened the issue if I hadn't. Back off, unless if I owe you an apology for filling out the Bug Report layout incorrectly. If I *did*  fill it out incorrectly, then perhaps y'all could make a minor amount of effort to make your fields a bit more descriptive. Had I known hardware was such an important part with this software, I would've been more verbose with my answers.

---

My hardware information:
![image](https://user-images.githubusercontent.com/46751150/158065443-c85f7121-4dcb-4596-8371-126281e133ee.png)
OpenGL Driver details (Used for rendering KDE, not recording): https://code.forsaken-borders.net/koteronavo.txt
CPU Info: https://code.forsaken-borders.net/ililevaqog.txt
GPU Info: 
![image](https://user-images.githubusercontent.com/46751150/158065952-689e9924-aff9-4bcf-9ea1-4e66896a8372.png)
![image](https://user-images.githubusercontent.com/46751150/158065956-2cca3e6e-bc81-458c-bdc8-f8f2c738829e.png)


Just to reiterate myself: OBS was functioning correctly and as intended 30 commits ago, however a recent commit led to it failing to use NVENC encoding. I do not have the time to track down which commit at the moment, which I sincerely apologize for. Thank you everyone for spending time on this issue.

**Comment by Fenrirthviti** [comment]
I wouldn't put too much stock in murphy's reply here, I have no idea who they are and they don't represent the project, and as far as I can tell this is the first time they've commented. Their reply is confusing and hard to follow as is.

That said, thanks for the additional context. I'm reasonable sure this is going to end up being some kind of environment issue, and it might be a good idea to swing by our linux-support channel in our [Discord](https://obsproject.com/discord) for someone a bit more familiar with the wonkiness that is NVIDIA drivers on Linux to take a look.

**Comment by WizardCM** [comment]
This bug was likely introduced in da4d98d0e46b60b993ff222a266354e6cff06417 and was fixed in 02e54103813f90eac34530de95c868251df01b9b. Pull latest and try again.

**Comment by OoLunar** [comment]
> I wouldn't put too much stock in murphy's reply here, I have no idea who they are and they don't represent the project, and as far as I can tell this is the first time they've commented. Their reply is confusing and hard to follow as is.

I see. Thank you for saying this.

> This bug was likely introduced in https://github.com/obsproject/obs-studio/commit/da4d98d0e46b60b993ff222a266354e6cff06417 and was fixed in https://github.com/obsproject/obs-studio/commit/02e54103813f90eac34530de95c868251df01b9b. Pull latest master and try again.

Latest commit works! Thank you everyone for the quick work.
